internal/rest/avi_obj_vs.go

```diff
@@ -560,24 +560,44 @@ func (rest *RestOperations) StatusUpdateForVS(restMethod utils.RestMethod, vsCac
 	serviceMetadataObj := vsCacheObj.ServiceMetadataObj
 	switch serviceMetadataObj.ServiceMetadataMapping("VS") {
 	case lib.GatewayVS:
-		updateOptions := status.UpdateOptions{
-			Vip:             IPAddrs,
-			ServiceMetadata: serviceMetadataObj,
-			Key:             key,
-			VSName:          vsCacheObj.Name,
-			Tenant:          vsCacheObj.Tenant,
-		}
-		statusOption := status.StatusOptions{
-			ObjType: lib.Gateway,
-			Op:      lib.UpdateStatus,
-			Key:     key,
-			Options: &updateOptions,
-		}
-		if lib.UseServicesAPI() {
-			statusOption.ObjType = lib.SERVICES_API
+		if vsCacheObj.ParentVSRef.Name != "" {
+			updateOptions := status.UpdateOptions{
+				ServiceMetadata: serviceMetadataObj,
+				Key:             key,
+				VSName:          vsCacheObj.Name,
+				Tenant:          vsCacheObj.Tenant,
+			}
+			statusOption := status.StatusOptions{
+				ObjType: lib.HTTPRoute,
+				Op:      lib.UpdateStatus,
+				Key:     key,
+				Options: &updateOptions,
+			}
+			if lib.UseServicesAPI() {
+				statusOption.ObjType = lib.SERVICES_API
+			}
+			utils.AviLog.Infof("key: %s Publishing to status queue, options: %v", key, utils.Stringify(statusOption))
+			status.PublishToStatusQueue(key, statusOption)
+		} else {
+			updateOptions := status.UpdateOptions{
+				Vip:             IPAddrs,
+				ServiceMetadata: serviceMetadataObj,
+				Key:             key,
+				VSName:          vsCacheObj.Name,
+				Tenant:          vsCacheObj.Tenant,
+			}
+			statusOption := status.StatusOptions{
+				ObjType: lib.Gateway,
+				Op:      lib.UpdateStatus,
+				Key:     key,
+				Options: &updateOptions,
+			}
+			if lib.UseServicesAPI() {
+				statusOption.ObjType = lib.SERVICES_API
+			}
+			utils.AviLog.Infof("key: %s Publishing to status queue, options: %v", updateOptions.ServiceMetadata.Gateway, utils.Stringify(statusOption))
+			status.PublishToStatusQueue(updateOptions.ServiceMetadata.Gateway, statusOption)
```

Comment:
We can refactor the above code to eliminate redeclaration - 
```
updateOptions := status.UpdateOptions{
        ServiceMetadata: serviceMetadataObj,
	Key:             key,
	VSName:          vsCacheObj.Name,
	Tenant:          vsCacheObj.Tenant,
}
if vsCacheObj.ParentVSRef.Name == "" {
    updateOptions.Vip = IPAddrs
}
statusOption := status.StatusOptions{
ObjType: lib.HTTPRoute,
Op:      lib.UpdateStatus,
Key:     key,
Options: &updateOptions,
}
if lib.UseServicesAPI() {
    statusOption.ObjType = lib.SERVICES_API
}
utils.AviLog.Infof("key: %s Publishing to status queue, options: %v", updateOptions.ServiceMetadata.Gateway, utils.Stringify(statusOption))
status.PublishToStatusQueue(updateOptions.ServiceMetadata.Gateway, statusOption)
```

Reply:
Done